content/how-to/code-splitting/splitting-vendor.md

```diff
@@ -0,0 +1,132 @@
+---
+title: How to Split Vendor Files?
+contributors:
+  - pksjce
+---
+
+A typical application uses third party libraries for framework/functionality needs. Particular versions of these libraries are used and code here does not change often. However, the application code changes frequently. Bundling application code with third party code would be inefficient. This is because the browser can cache asset files based on the cache header and files can be cached without needing to call the cdn again if it's contents don't change. To take advantage of this, we want to keep the hash of the vendor files constant regardless of application code changes. We can do this only when we separate the bundles for vendor and application code.
+
+Let's consider a sample application, that uses [momentjs](https://www.npmjs.com/package/moment) which is a time formatting library commonly used.
+
+Install `moment` as follows in your application directory.
+
+`npm install --save moment`
+
+The index file will require `moment` as a dependency and log the current date as follows
+
+```javascript
+\\ index.js
+
+var moment = require('moment');
+console.log(moment().format())
+
+```
+
+We can bundle the application with webpack using the following config
+
+```javascript
+\\ webpack.config.js
+
+modules.export = function(env) {
+    return {
+        entry: './index.js',
+        output: {
+            filename: '[chunkhash].[name].js`,
+            path: './dist'
+        }
+    }
+}
+```
+
+On running `webpack` in your application, if you inspect the resulting bundle, you will see that `moment` and `index.js` have been bundled in `bundle.js`.
+
+This is not ideal for the application. If the code in `index.js` changes, then the whole bundle is rebuilt. The browser will have to load a new copy of the new bundle even though most of it hasn't changed at all.
+
+### Multiple entries
+
+Let's try to mitigate this by adding a separate entry point for `moment` and name it `vendor`
+
+```javascript
+\\ webpack.config.js
+
+modules.export = function(env) {
+    return {
+        entry: {
+            main: './index.js',
+            vendor: 'moment'
+        },
+        output: {
+            filename: '[chunkhash].[name].js`,
+            path: './dist'
+        }
+    }
+}
+```
+
+On running `webpack` now, we see that two bundles have been created. If you inspect these though, you will find that the code for `moment` is present in both the files!
+
+It is for this reason, that we will need to use the [CommonsChunkPlugin](https://github.com/webpack/webpack/blob/master/lib/optimize/CommonsChunkPlugin.js).
```

Comment:
This link can break if the source changes. Maybe link to a hash (specific commit). Not sure.